data/modifications/pick_advance/dialog.lua

```diff
@@ -0,0 +1,168 @@
+-- << pickadvance_dialog
+
+pickadvance = {}
+local pickadvance = pickadvance
+local wesnoth = wesnoth
+local ipairs = ipairs
+local table = table
+local T = wesnoth.require("lua/helper.lua").set_wml_tag_metatable {}
+local translate = wesnoth.textdomain "wesnoth"
+
+local function filter_false(arr)
+	local result = {}
+	for _, v in ipairs(arr) do
+		if v ~= false then
+			result[#result + 1] = v
+		end
+	end
+	return result
+end
+
+
+function pickadvance.show_dialog_unsynchronized(advance_info, unit)
+-- dialog exit codes --
+  local help_code = -4
+  local reset_code = -3
+  local single_unit_code = -1
+  local all_units_code = 1
+--    
+	local label = "Plan advance:"
+
+	local unit_type_options = advance_info.type_advances
+	--print_as_json("advances for", unit.type, unit_type_options)
+	local options = {}
+	for _, ut in ipairs(unit_type_options) do
+		options[#options + 1] = wesnoth.unit_types[ut]
+	end
+	local show_images = true
+
+	local unit_override_one = (advance_info.unit_override or {})[2] == nil
+		and (advance_info.unit_override or {})[1] or nil
+	local game_override_one = (advance_info.game_override or {})[2] == nil
+		and (advance_info.game_override or {})[1] or nil
+
+	local description_row = T.row {
+		T.column { T.label { use_markup = true, label = label } },
+	}
+
+	local list_sub_row
+	if show_images then
+		list_sub_row = T.row {
+			T.column { T.image { id = "the_icon" } },
+			T.column { grow_factor = 0, T.label { use_markup = true, id = "the_label" } },
+			T.column { grow_factor = 1, T.spacer {} },
+		}
+	else
+		list_sub_row = T.row {
+			T.column { horizontal_alignment = "left", T.label { use_markup = true, id = "the_label" } }
+		}
+	end
+
+	local toggle_panel = T.toggle_panel { return_value = single_unit_code, T.grid { list_sub_row } }
+
+	local list_definition = T.list_definition { T.row { T.column { horizontal_grow = true, toggle_panel } } }
+
+	local listbox = T.listbox { id = "the_list", list_definition, has_minimum = true }
+
+	local reset_button = T.button {
+		return_value = reset_code,
+		label = "\n" .. translate("Reset") .. "\n",
+		tooltip = "Reset advancements to default"
```

Comment:
I wonder if the quality of the addon is sufficient for Wesnoth's quality standards. I mean, this text is untranslated. I never figured out how to do that properly and efficiently. I think adding the addon to mainline would still be a clear improvement over the current situation, just making the choice explicit here.

Reply:
Oh sorry, I now see that this question was raised by @Pentarctagon already. Resolving it here to de-dup.